x-pack/plugins/cases/server/common/utils.ts

```diff
@@ -398,3 +414,74 @@ export const getNoneCaseConnector = () => ({
   type: ConnectorTypes.none,
   fields: null,
 });
+
+interface LensMarkdownNode {
+  timeRange: TimeRange;
+  editMode: boolean;
+  attributes: LensDocShape714 & { references: SavedObjectReference[] };
+}
+
+export const parseCommentString = (comment: string) => {
+  const processor = unified().use([[markdown, {}], LensParser, TimelineParser]);
+  return processor.parse(comment) as Parent;
+};
+
+export const stringifyComment = (comment: Parent) =>
+  unified()
+    .use([
+      [
+        remarkStringify,
+        {
+          allowDangerousHtml: true,
+          handlers: {
+            /*
+              because we're using rison in the timeline url we need
+              to make sure that markdown parser doesn't modify the url
+            */
+            timeline: TimelineSerializer,
+            lens: LensSerializer,
+          },
+        },
+      ],
+    ])
+    .stringify(comment);
+
+export const getLensVisualizations = (parsedComment: Array<LensMarkdownNode | Node>) =>
+  filter(parsedComment, { type: LENS_ID }) as LensMarkdownNode[];
+
+export const extractLensReferencesFromCommentString = (comment: string): SavedObjectReference[] => {
+  const parsedComment = parseCommentString(comment);
+  const lensVisualizations = getLensVisualizations(parsedComment.children);
+  const flattenRefs = flatMap(lensVisualizations, (lensObject) => lensObject.attributes.references);
```

Comment:
even thou lens SO is easy to parse you should not do it, as it might change in the future. Lens needs to export functions to extract and inject references from/to their state which you should use here.

Reply:
we will be happy to work with the Lens team to help with providing this kind of API